JWBestPracticeApps/Offline DRM/Offline DRM/ViewController.swift

```diff
@@ -0,0 +1,206 @@
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Create the configuration for the AVAssetDownloadURLSession.
+        let backgroundConfiguration = URLSessionConfiguration.background(withIdentifier: "AAPL-Identifier")
+        assetDownloadURLSession = AVAssetDownloadURLSession(configuration: backgroundConfiguration,
+                                                            assetDownloadDelegate: self, delegateQueue: delegateQueue)
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        guard let playerViewController = segue.destination as? JWPlayerViewController else {
+            return
+        }
+        // Set this controller's player reference to the embedded JWPlayerViewController's.
+        self.player = playerViewController.player
+    }
+    
+    func getPlaylist(remote url: URL) -> JWPlayerConfiguration? {
+        let sem = DispatchSemaphore(value: 0)
+        let task = URLSession.shared.dataTask(with: url) { [weak self] data, response, err in
+            defer {
+                sem.signal()
+            }
+            guard let self = self else {
+                return
+            }
+            if let error = err {
+                fatalError("Playlist URL request failed: \(error.localizedDescription)")
+            }
+            do {
+                let playlist = try JSONDecoder().decode(DRMPlaylist.self, from: data!)
+                guard let source = playlist.playlist.first?.sources.first(where: { $0.type == "application/vnd.apple.mpegurl" }) else {
+                    return
+                }
+                // After parsing the playlist, set the current video file
+                self.videoFile = source.file
+                // Set the certificate URL we will need for for the JWDRMKeyDataSource
+                self.keyDataSource.certificateURLStr = source.drm.fairplay?.certificateURL
+                // Set the process SPC URL we will need for for the JWDRMKeyDataSource
+                self.keyDataSource.processSPCURLStr = source.drm.fairplay?.processSpcURL
+            } catch {
+                print("Error decoding data from playlist URL", error.localizedDescription)
+            }
+        }
+        task.resume()
+        sem.wait()
+        // Create a configuration from the video file.
+        let fileURL = URL(string: videoFile)!
+        var config: JWPlayerConfiguration? = nil
+        do {
+            let playerItem = try JWPlayerItemBuilder().file(fileURL).build()
+            config = try JWPlayerConfigurationBuilder()
+                .playlist(items: [playerItem]).build()
+        } catch {
+            // Handle any error thrown during the configuration build
+            print("Error encountered while building the configuration:", error.localizedDescription)
+        }
+       
+        return config
+    }
+    
+    func getPlaylist(local url: URL) -> JWPlayerConfiguration? {
```

Comment:
Same here. Some header doc since this is public.

Reply:
addressed